app/app/vendor/routes.js

```diff
@@ -0,0 +1,9 @@
+module.exports = function () {
+  angular.module('app.routes', ['ngRoute']).config(['$routeProvider', function ($routeProvider) {
+    $routeProvider
+      .when('/', require('./routes/index').home)
+      .when('/permission-denied', require('./routes/index').permission_denied)
```

Comment:
Identifier 'permission_denied' is not in camel case.<br>'require' is not defined.